CI failure: Deploy — Missing required credentials for al push command

## CI Failure Report

**Workflow:** Deploy
**Branch:** main
**Commit:** d3e377f9a81925850a580d5890d2a89890b25e9b
**Run:** https://github.com/Action-Llama/agents/actions/runs/23413951777

## Failure Summary

The deployment workflow is failing because the `npx al push` command cannot find required credentials (github_token, git_ssh, anthropic_key) despite the `--no-creds` flag being used.

## Failed Job(s)

### deploy

**Error output:**
```
  [MISSING] GitHub Personal Access Token (github_token)
  [MISSING] SSH Key & Git Identity (git_ssh)
  [MISSING] Anthropic API Credential (anthropic_key)
Credential error: 3 credential(s) missing: github_token, git_ssh, anthropic_key.
Run 'al doctor' interactively to configure them.
##[error]Process completed with exit code 1.
```

## Root Cause Analysis

The `al push` command from `@action-llama/action-llama` requires three credentials to be configured:
1. **github_token** - GitHub Personal Access Token
2. **git_ssh** - SSH Key & Git Identity 
3. **anthropic_key** - Anthropic API Credential

The current workflow only sets up:
- An SSH key at `~/.ssh/deploy_key` (from `DEPLOY_SSH_KEY` secret)
- A configuration file at `~/.action-llama/environments/prod.toml` (from `DEPLOY_ENV_TOML` secret)

However, the `al` command expects these credentials to be configured in a different format or location. The `--no-creds` flag appears to mean "don't prompt for credentials interactively" rather than "don't require credentials at all".

## Suggested Fix

The workflow needs to properly configure the required credentials for the `al` command. Here are the possible solutions:

1. **Update the environment setup to provide credentials in the format expected by `al`:**
   - Set environment variables for the credentials (e.g., `GITHUB_TOKEN`, `GIT_SSH_KEY`, `ANTHROPIC_API_KEY`)
   - Or ensure the `prod.toml` file contains all required credential configurations

2. **Update the Deploy step to properly configure credentials before running `al push`:**
   ```diff
   - name: Deploy
   - run: npx al push --env prod --headless --no-creds
   + run: |
   +   # Configure credentials for al command
   +   export GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
   +   export GIT_SSH_COMMAND="ssh -i ~/.ssh/deploy_key -o StrictHostKeyChecking=no"
   +   export ANTHROPIC_API_KEY=${{ secrets.ANTHROPIC_API_KEY }}
   +   npx al push --env prod --headless --no-creds
   ```

3. **Investigate if the `prod.toml` configuration file should contain the credential specifications** and ensure `DEPLOY_ENV_TOML` secret contains the proper format.

## Additional Context

- This workflow was introduced in commit e6be662 on March 22, 2026
- The repository recently switched from using GitHub source to npm registry for `@action-llama/action-llama` dependency (currently at version ^0.14.1)
- There have been multiple recent CI-related fixes, suggesting ongoing deployment configuration issues

---
*This issue was automatically created by the gh-actions-responder agent.*

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -37,5 +37,21 @@ jobs:
           mkdir -p ~/.action-llama/environments
           echo '${{ secrets.DEPLOY_ENV_TOML }}' > ~/.action-llama/environments/prod.toml
 
+      - name: Configure credentials for action-llama
+        run: |
+          # Set up the credential directory
+          mkdir -p ~/.action-llama/credentials
+          
+          # Configure GitHub token
+          echo '{"type":"github_token","token":"${{ secrets.GITHUB_TOKEN }}"}' > ~/.action-llama/credentials/github_token.json
+          
+          # Configure SSH/Git credentials - use the deploy key  
+          GIT_EMAIL="${{ vars.GIT_EMAIL }}"
+          GIT_NAME="${{ vars.GIT_NAME }}"
+          echo '{"type":"git_ssh","privateKey":"${{ secrets.DEPLOY_SSH_KEY }}","email":"'${GIT_EMAIL:-deploy@action-llama.com}'","name":"'${GIT_NAME:-Action Llama Deploy}'"}' > ~/.action-llama/credentials/git_ssh.json
+          
+          # Configure Anthropic API key 
+          echo '{"type":"anthropic_key","key":"${{ secrets.ANTHROPIC_API_KEY }}"}' > ~/.action-llama/credentials/anthropic_key.json
+
       - name: Deploy
         run: npx al push --env prod --headless --no-creds

